modules/imgproc/src/fixedpoint.inl.hpp

```diff
@@ -91,9 +91,14 @@ class fixedpoint64
     CV_ALWAYS_INLINE fixedpoint64 operator << (int n) const { return fixedpoint64(val << n); }
     CV_ALWAYS_INLINE bool operator == (const fixedpoint64& val2) const { return val == val2.val; }
     template <typename ET>
-    CV_ALWAYS_INLINE operator ET() const { return cv::saturate_cast<ET>((int64_t)fixedround((uint64_t)val) >> fixedShift); }
+    CV_ALWAYS_INLINE ET saturate_cast() const { return cv::saturate_cast<ET>((int64_t)fixedround((uint64_t)val) >> fixedShift); }
     CV_ALWAYS_INLINE operator double() const { return (double)val / (1LL << fixedShift); }
     CV_ALWAYS_INLINE operator float() const { return (float)val / (1LL << fixedShift); }
+    CV_ALWAYS_INLINE operator uint8_t() const { return saturate_cast<uint8_t>(); }
+    CV_ALWAYS_INLINE operator int8_t() const { return saturate_cast<int8_t>(); }
+    CV_ALWAYS_INLINE operator uint16_t() const { return saturate_cast<uint16_t>(); }
+    CV_ALWAYS_INLINE operator int16_t() const { return saturate_cast<int16_t>(); }
+    CV_ALWAYS_INLINE operator int32_t() const { return saturate_cast<int32_t>(); }
```

Comment:
IMHO it seems somewhat insymmetrical: int8_t/uint8_t, int16_t/uint16_t, int32_t/(void)